modules/standard/Time.chpl

```diff
@@ -678,6 +688,7 @@ enum day       { sunday=0, monday, tuesday, wednesday, thursday, friday, saturda
     this.chpl_second = second;
     this.chpl_microsecond = microsecond;
     this.chpl_tz = tz;
+    this.tz_aware = true;
```

Comment:
if `tz == nil`, is this `tz_aware`?

Reply:
That's unfortunate. Can that lead to problems now that you have removed some of the check for nils?

I see you left `proc dateTime.utcOffset()` with nil runtime check. It should probably have a compile time check, but you probably need to also check for nil at runtime if a user creates a `tz_aware` `dateTime` and then calls this method

Reply:
I didn't think it could since after the (removed) nil check in the operators, we work with instances that have different timezones. But actually, we do need to do use them for `utcOffset` in that case, or the result is meaningless.

I didn't worry about `dateTime.utcOffset` (and `dateTime.dst`) in this PR since they are soon to be deprecated, but didn't think